### ISSUE TITLE
IBD should write a sample exclusion list to the global annotations

### ISSUE THREAD

**Issue by danking** [hail-is/hail#863]
IBD should be expanded with two alternative output formats.
## Greedy by Degree

```
ibd --exclusion-list [ --min FLOAT ] [ --max FLOAT ]

ibd --exclusion-list --min 0.2 --max 0.9
```

This invocation uses a greedy algorithm to eliminate samples which have a `pi_hat` at or greater than the minimum and at or below the maximum with at least one other sample. The algorithm is essentially:
1. compute IBD `pi_hat` matrix
2. create a graph whose nodes are samples and whose edges from `s1` to `s2` exist if and only if the `pi_hat` for `s1` and `s2` is greater than the threshold.
3. remove the sample with highest [degree](https://en.wikipedia.org/wiki/Degree_%28graph_theory%29) and all edges involving it
4. unless the graph is completely disconnected (i.e. no remaining samples have `pi_hat` greater than the threshold with any remaining samples), go to step 2
5. write the list of removed samples to the global annotation `ga.ibdFilteredSamples`
## Greedy by Custom Weight

```
ibd --exclusion-list-with-weight EXPRESSION [ --min FLOAT ] [ --max FLOAT ]

ibd --exclusion-list-with-weight 'sa.goodness[s]' --min 0.85
```

This invocation uses the same greedy algorithm as above except that step 3 removes the sample with highest weight according to the given expression. There is only one identifier in the scope of that expression: `s`, the sample id.

**Comment by danking** [comment]
@jbloom22 notes that we're solving the maximal clique problem on the complement of the graph. Perhaps we can apply some known approximation algorithms.

**Comment by jbloom22** [comment]
More directly, this is known as the maximal independent set (MIS) problem, for which there are simple parallel algorithms: https://cstar.iiit.ac.in/~kkishore/MISStudy.pdf

MIS should be implemented in Spark GraphX, but I can't find it!
http://ilpubs.stanford.edu:8090/1085/2/primitives_tr_sig_alternate.pdf

**Comment by jbloom22** [comment]
@johnc1231 update this one your PR is in.

**Comment by danking** [comment]
@johnc1231 If your *commit message* (not the PR) contains `fixes #863` you should be good.

**Comment by tpoterba** [comment]
We added a maximal independent set function to key table
